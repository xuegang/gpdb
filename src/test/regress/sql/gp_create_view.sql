-- start_ignore
drop table if exists sourcetable cascade;
drop view if exists v_sourcetable cascade;
drop view if exists v_sourcetable1 cascade;
-- end_ignore

create table sourcetable 
(
        cn int not null,
        vn int not null,
        pn int not null,
        dt date not null,
        qty int not null,
        prc float not null,
        primary key (cn, vn, pn)
) distributed by (cn,vn,pn);

insert into sourcetable values
  ( 2, 40, 100, '1401-1-1', 1100, 2400),
  ( 1, 10, 200, '1401-3-1', 1, 0),
  ( 3, 40, 200, '1401-4-1', 1, 0),
  ( 1, 20, 100, '1401-5-1', 1, 0),
  ( 1, 30, 300, '1401-5-2', 1, 0),
  ( 1, 50, 400, '1401-6-1', 1, 0),
  ( 2, 50, 400, '1401-6-1', 1, 0),
  ( 1, 30, 500, '1401-6-1', 12, 5),
  ( 3, 30, 500, '1401-6-1', 12, 5),
  ( 3, 30, 600, '1401-6-1', 12, 5),
  ( 4, 40, 700, '1401-6-1', 1, 1),
  ( 4, 40, 800, '1401-6-1', 1, 1);

create view  v_sourcetable as select * from sourcetable order by vn;
select * from v_sourcetable;

create view v_sourcetable1 as SELECT sourcetable.qty, vn, pn FROM sourcetable union select sourcetable.qty, sourcetable.vn, sourcetable.pn from sourcetable order by qty;
select * from v_sourcetable1;
