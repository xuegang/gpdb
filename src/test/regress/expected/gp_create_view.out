-- start_ignore
drop table if exists sourcetable cascade;
NOTICE:  table "sourcetable" does not exist, skipping
drop view if exists v_sourcetable cascade;
NOTICE:  view "v_sourcetable" does not exist, skipping
drop view if exists v_sourcetable1 cascade;
NOTICE:  view "v_sourcetable1" does not exist, skipping
-- end_ignore
create table sourcetable 
(
        cn int not null,
        vn int not null,
        pn int not null,
        dt date not null,
        qty int not null,
        prc float not null,
        primary key (cn, vn, pn)
) distributed by (cn,vn,pn);
NOTICE:  CREATE TABLE / PRIMARY KEY will create implicit index "sourcetable_pkey" for table "sourcetable"
insert into sourcetable values
  ( 2, 40, 100, '1401-1-1', 1100, 2400),
  ( 1, 10, 200, '1401-3-1', 1, 0),
  ( 3, 40, 200, '1401-4-1', 1, 0),
  ( 1, 20, 100, '1401-5-1', 1, 0),
  ( 1, 30, 300, '1401-5-2', 1, 0),
  ( 1, 50, 400, '1401-6-1', 1, 0),
  ( 2, 50, 400, '1401-6-1', 1, 0),
  ( 1, 30, 500, '1401-6-1', 12, 5),
  ( 3, 30, 500, '1401-6-1', 12, 5),
  ( 3, 30, 600, '1401-6-1', 12, 5),
  ( 4, 40, 700, '1401-6-1', 1, 1),
  ( 4, 40, 800, '1401-6-1', 1, 1);
create view  v_sourcetable as select * from sourcetable order by vn;
select * from v_sourcetable;
 cn | vn | pn  |     dt     | qty  | prc  
----+----+-----+------------+------+------
  1 | 10 | 200 | 03-01-1401 |    1 |    0
  1 | 20 | 100 | 05-01-1401 |    1 |    0
  3 | 30 | 600 | 06-01-1401 |   12 |    5
  1 | 30 | 500 | 06-01-1401 |   12 |    5
  1 | 30 | 300 | 05-02-1401 |    1 |    0
  3 | 30 | 500 | 06-01-1401 |   12 |    5
  4 | 40 | 700 | 06-01-1401 |    1 |    1
  2 | 40 | 100 | 01-01-1401 | 1100 | 2400
  4 | 40 | 800 | 06-01-1401 |    1 |    1
  3 | 40 | 200 | 04-01-1401 |    1 |    0
  2 | 50 | 400 | 06-01-1401 |    1 |    0
  1 | 50 | 400 | 06-01-1401 |    1 |    0
(12 rows)

create view v_sourcetable1 as SELECT sourcetable.qty, vn, pn FROM sourcetable union select sourcetable.qty, sourcetable.vn, sourcetable.pn from sourcetable order by qty;
select * from v_sourcetable1;
 qty  | vn | pn  
------+----+-----
    1 | 20 | 100
    1 | 40 | 200
    1 | 40 | 700
    1 | 10 | 200
    1 | 40 | 800
    1 | 50 | 400
    1 | 30 | 300
   12 | 30 | 600
   12 | 30 | 500
 1100 | 40 | 100
(10 rows)

